src/main/kotlin/blackjack/domain/Dealer.kt

```diff
@@ -1,16 +1,31 @@
 package blackjack.domain
 
 class Dealer : Participant("딜러") {
-    override fun isDealer(): Boolean {
-        return true
-    }
+    override fun isDealer() = true
 
     override fun canNotReceiveCard(): Boolean {
         return score > MAXIMUM_SCORE_TO_RECEIVE_CARD
     }
 
-    fun determineResult(gamblerResults: List<GamblerResult>): DealerResult {
-        return DealerResult(this, gamblerResults)
+    fun determineResult(gamblers: List<Gambler>): BlackjackResults {
+        val gamblerResults = determineGamblerResults(gamblers)
+        val dealerResult = determineDealerResult(gamblerResults)
+        return BlackjackResults(dealerResult, gamblerResults)
+    }
+
+    private fun determineGamblerResults(gamblers: List<Gambler>): List<GamblerResult> {
+        val gamblerResults = gamblers.map { gambler ->
+            val resultStatus = ResultStatus.of(gambler, this)
+            GamblerResult(gambler, gambler.calculateProfit(resultStatus.profitRate))
+        }
+        return gamblerResults
```

Comment:
생각해보니 변수 선언이 불필요해보이네요. 😃
```suggestion
        return gamblers.map { gambler ->
            val resultStatus = ResultStatus.of(gambler, this)
            GamblerResult(gambler, gambler.calculateProfit(resultStatus.profitRate))
        }
```